### DISCUSSION TITLE
Texture - Converting circular structure to JSON

### DISCUSSION THREAD

**Discussion by wpitallo** [pixijs/pixijs#7953] [Ideas]
Hi, 

Just wandering why we cannot serialize a texture object, could this not also cause other issues maybe?

![image](https://user-images.githubusercontent.com/12214559/140197739-14f8f110-1467-443c-a03a-c8c50e63b138.png)

Note: added:  
```
let test = JSON.stringify(texture)
```
![image](https://user-images.githubusercontent.com/12214559/140197794-ecfa928b-499c-48d0-a3ac-c8fe311e5c5d.png)

### Environment

- **`pixi.js` version**: _e.g. 6.2.0_
- **Browser & Version**: _e.g. Chrome 67_
- **OS & Version**: _e.g. All_
- **Running Example**: https://pixijs.io/examples/#/demos-basic/container.js

**Comment by bigtimebuddy** [comment]
Serializing Pixi objects is not supported. You'll need to come up with your own solution. Unless other folks have recommendations.

**Comment by miltoncandelero** [comment]
The biggest offender if I remember correctly was ObservablePoint.
It keeps a reference to the context of the function it's supposed to call when it changes and that usually is the same object that has the point to begin with.
Try using a custom `replacer` and `reviver` in the `JSON.parse` method or try something like https://www.npmjs.com/package/flatted

**Comment by wpitallo** [comment]
Kinda still seems like a bug / not ideal to have any kind of circular dependencies. Thanx for the workaround tho will try give them a try 🙂. Should the first prize not be to remove the circular dependencies tho?

**Comment by bigtimebuddy** [comment]
I'm moving this to a discussion. Supporting JSON serialization is not a feature we are interested in adding, but someone could easily come up with a plugin solution to mixin `toJSON` and `fromJSON` methods to relevant display objects.
